jdaviz/core/template_mixin.py

```diff
@@ -1942,6 +1946,244 @@ def __init__(self, *args, **kwargs):
+            # assume first dataset (for retrieving the region object)
+            # but iterate over all subsets
+            spatial_regions = [self._get_spatial_region(dataset=self.dataset.selected[0], subset=subset)  # noqa
+                               for subset in self.selected if subset != self._manual_options]
+        else:
+            # use cached version
+            spatial_regions = [self.selected_spatial_region]
+
+        x_coords, y_coords = np.array([]), np.array([])
+        for spatial_region in spatial_regions:
+            if spatial_region is None:
+                continue
+
+            if isinstance(spatial_region, PixelRegion):
+                pixel_region = spatial_region
+            else:
+                wcs = getattr(viewer.state.reference_data, 'coords', None)
+                if wcs is None:
+                    return [], []
+                pixel_region = spatial_region.to_pixel(wcs)
+            roi = regions2roi(pixel_region)
+
+            # NOTE: this assumes that we'll apply the same radius factor to all subsets (all will
+            # be defined at the same slice for cones in cubes)
+            if hasattr(roi, 'radius'):
+                roi.radius *= self.scale_factor
+            elif hasattr(roi, 'radius_x'):
+                roi.radius_x *= self.scale_factor
+                roi.radius_y *= self.scale_factor
+            elif hasattr(roi, 'center'):
+                center = roi.center()
+                width_orig = roi.xmax - roi.xmin
+                height_orig = roi.ymax - roi.ymin
+                roi.xmin = center[0] - width_orig/2 * self.scale_factor
+                roi.xmax = center[0] + width_orig/2 * self.scale_factor
+                roi.ymin = center[1] - height_orig/2 * self.scale_factor
+                roi.ymax = center[1] + height_orig/2 * self.scale_factor
+            else:  # pragma: no cover
+                raise NotImplementedError
+
+            x, y = roi.to_polygon()
+
+            # concatenate with nan between to avoid line connecting separate subsets
+            x_coords = np.concatenate((x_coords, np.array([np.nan]), x))
+            y_coords = np.concatenate((y_coords, np.array([np.nan]), y))
+
+        return x_coords, y_coords
+
+    def _update_mark_coords(self, *args):
+        for viewer in self.image_viewers:
+            x_coords, y_coords = self._get_mark_coords(viewer)
+            for mark in self.marks:
+                mark.x, mark.y = x_coords, y_coords
+
+
+class ApertureSubsetSelectMixin(VuetifyTemplate, HubListener):
+    """
+    Applies the SubsetSelect component as a mixin in the base plugin.  This
+    automatically adds traitlets as well as new properties to the plugin with minimal
+    extra code.  For multiple instances or custom traitlet names/defaults, use the
+    component instead.
+
+    To use in a plugin:
+
+    * add ``ApertureSubsetSelectMixin`` as a mixin to the class BEFORE ``DatasetSelectMixin``
```

Comment:
Maybe explain here why *BEFORE* is emphasize here?

Reply:
because it is necessary to add it before and not after because of inheritance order